### ISSUE TITLE
migrate to meson 

### ISSUE THREAD

**Issue by korli** [haikuports/haikuports#4371]
- [ ] libnice
- [ ] fribidi
- [ ] gst-plugins-bad
- [ ] gst-plugins-good
- [ ] gst-plugins-ugly
- [ ] p11-kit
- [ ] pango
- [ ] pixman
- [ ] radare2

**Comment by korli** [comment]
gtk-doc done in 6669c308548f7672198ffab11986f9acfcec0baa

**Comment by korli** [comment]
libsoup3 done in b420b3f6138528c688322fcba6ce4be04ef2b405

**Comment by korli** [comment]
pango done in 8609879a5efcfd8949f0a58424e177c0f7d69284

**Comment by korli** [comment]
fribidi done in c4fab2d118eff3f60a656d54d7896e01a2322e13

**Comment by Begasus** [comment]
@korli, so far got a package done for radare2 with configure, a meson fix has been provided (need to test) for building with meson,  @trufae supplied it here: https://github.com/radareorg/radare2/commit/19833df99b4bfa54a9ab295d3ac599bd15ea75ca

**Comment by korli** [comment]
@Begasus are system dependencies correctly found and enabled? xxhash, capstone, etc

**Comment by trufae** [comment]
Please dont link against system capstone. Last release was 6 years ago and it doesnt handle any of the recent instructions used by modern compilers like all the spectra stuff, pac, etc.

about xxhash its not really used, same goes for openssl, nobody does https:/: in r2, and also for libuv, no need to link against those libs. About libmagic, r2 one is faster and works better than gnu one, but its fine to link against them but carving memory with the system libmagic is slow

**Comment by Begasus** [comment]
> @Begasus are system dependencies correctly found and enabled? xxhash, capstone, etc

@korli yes, everything detected and enabled, but maybe as @trufae mentions leave them out? (seen some passing by for those on an issue upstream too iirc).
PS, doing a meson checkout with latest changes upstream fixed the build, install part is now failing because it wants to install the scons (python) to /boot/... (read-only, so need to check if I can change the path for that)

**Comment by Begasus** [comment]
> Please dont link against system capstone. Last release was 6 years ago

Last release (4.0.2) was released in 2020, if I'm not wrong this is only 3 years ago? ;)

**Comment by trufae** [comment]
Capstone5 existed in a branch already in the time. I'm thinking about capstone4 as the 4.0.0 which was 5-6 years ago. And 4.0.2 just fixed minor bugs. Didnt introduced any new instructionset or architecture.

**Comment by Begasus** [comment]
Yeah, been reading up on some posts about it (Gentoo does seem to go that direction too).
In the meantime the patch for ptrace didn't really solve the build:
```
io.c:(.text+0x221): undefined reference to `ptrace'
```

**Comment by Begasus** [comment]
> > io.c
> 
> should probably add haiku to https://github.com/radareorg/radare2/blob/6bb0208c28f2788016ea56ba2ad157a0154ff78c/meson.build#L324

Already done (and was my first attempt on it yesterday too), that worked. :)

**Comment by korli** [comment]
> **EDIT2** successful build with meson now :)

Could you check if https://github.com/radareorg/radare2/commit/986f7c75e8c67ed01de406e8a8132ac0e712e8bc can be reverted? it is in a #ifdef HAVE_PTRACE

**Comment by Begasus** [comment]
> > **EDIT2** successful build with meson now :)
> 
> Could you check if [radareorg/radare2@986f7c7](https://github.com/radareorg/radare2/commit/986f7c75e8c67ed01de406e8a8132ac0e712e8bc) can be reverted? it is in a #ifdef HAVE_PTRACE

Already reverted, fixed the package so I don't get the message on iaito not finding the themes also, finetuning the recipe before I push here. :)

**Comment by Begasus** [comment]
radare2 updated in: https://github.com/haikuports/haikuports/pull/8864 :)
